From c7dea12b7bccb6c96f02c37d3cf3aab755b8014a Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@gmail.com>
Date: Sun, 26 Oct 2014 22:05:31 +0100
Subject: Fix build on Haiku.

- Look for socket in libnetwork
- Look for forkpty in libbsd

diff --git a/configure.ac b/configure.ac
index a0857d3..f500a1f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -179,7 +179,7 @@ AS_IF([test x"$with_utempter" != xno],
 
 AC_SEARCH_LIBS([compress], [z], , [AC_MSG_ERROR([Unable to find zlib.])])
 
-AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([socket], [socket network])
 AC_SEARCH_LIBS([inet_addr], [nsl])
 
 AC_SEARCH_LIBS([clock_gettime], [rt])
@@ -352,10 +352,8 @@ AC_CHECK_DECL([forkpty],
   , [[#include <sys/types.h>
   #include <libutil.h>]])
 
-AC_CHECK_LIB([util], [forkpty], [
+AC_SEARCH_LIBS([forkpty], [util bsd], [
   AC_DEFINE([HAVE_FORKPTY],, [Define if you have forkpty().])
-  LIBUTIL="-lutil"
-  AC_SUBST([LIBUTIL])
 ])
 
 AC_MSG_CHECKING([whether pipe2(..., O_CLOEXEC) is supported])
diff --git a/src/examples/Makefile.am b/src/examples/Makefile.am
index faa2286..a8c664f 100644
--- a/src/examples/Makefile.am
+++ b/src/examples/Makefile.am
@@ -15,16 +15,16 @@ decrypt_LDADD = ../crypto/libmoshcrypto.a $(CRYPTO_LIBS)
 
 parse_SOURCES = parse.cc
 parse_CPPFLAGS = -I$(srcdir)/../terminal -I$(srcdir)/../util
-parse_LDADD = ../terminal/libmoshterminal.a ../util/libmoshutil.a $(LIBUTIL)
+parse_LDADD = ../terminal/libmoshterminal.a ../util/libmoshutil.a
 
 termemu_SOURCES = termemu.cc
 termemu_CPPFLAGS = -I$(srcdir)/../terminal -I$(srcdir)/../util -I$(srcdir)/../statesync -I../protobufs
-termemu_LDADD = ../terminal/libmoshterminal.a ../util/libmoshutil.a ../statesync/libmoshstatesync.a ../protobufs/libmoshprotos.a $(LIBUTIL) $(TINFO_LIBS) $(protobuf_LIBS)
+termemu_LDADD = ../terminal/libmoshterminal.a ../util/libmoshutil.a ../statesync/libmoshstatesync.a ../protobufs/libmoshprotos.a $(TINFO_LIBS) $(protobuf_LIBS)
 
 ntester_SOURCES = ntester.cc
 ntester_CPPFLAGS = -I$(srcdir)/../util -I$(srcdir)/../statesync -I$(srcdir)/../terminal -I$(srcdir)/../network -I$(srcdir)/../crypto -I../protobufs $(protobuf_CFLAGS)
-ntester_LDADD = ../statesync/libmoshstatesync.a ../terminal/libmoshterminal.a ../network/libmoshnetwork.a ../crypto/libmoshcrypto.a ../protobufs/libmoshprotos.a ../util/libmoshutil.a $(LIBUTIL) -lm $(protobuf_LIBS)  $(CRYPTO_LIBS)
+ntester_LDADD = ../statesync/libmoshstatesync.a ../terminal/libmoshterminal.a ../network/libmoshnetwork.a ../crypto/libmoshcrypto.a ../protobufs/libmoshprotos.a ../util/libmoshutil.a -lm $(protobuf_LIBS)  $(CRYPTO_LIBS)
 
 benchmark_SOURCES = benchmark.cc
 benchmark_CPPFLAGS = -I$(srcdir)/../util -I$(srcdir)/../statesync -I$(srcdir)/../terminal -I../protobufs -I$(srcdir)/../frontend -I$(srcdir)/../crypto -I$(srcdir)/../network $(protobuf_CFLAGS)
-benchmark_LDADD = ../frontend/terminaloverlay.o ../statesync/libmoshstatesync.a ../terminal/libmoshterminal.a ../protobufs/libmoshprotos.a ../network/libmoshnetwork.a ../crypto/libmoshcrypto.a ../util/libmoshutil.a $(STDDJB_LDFLAGS) $(LIBUTIL) -lm $(TINFO_LIBS) $(protobuf_LIBS) $(CRYPTO_LIBS)
+benchmark_LDADD = ../frontend/terminaloverlay.o ../statesync/libmoshstatesync.a ../terminal/libmoshterminal.a ../protobufs/libmoshprotos.a ../network/libmoshnetwork.a ../crypto/libmoshcrypto.a ../util/libmoshutil.a $(STDDJB_LDFLAGS) -lm $(TINFO_LIBS) $(protobuf_LIBS) $(CRYPTO_LIBS)
diff --git a/src/frontend/Makefile.am b/src/frontend/Makefile.am
index 5856469..a9d4188 100644
--- a/src/frontend/Makefile.am
+++ b/src/frontend/Makefile.am
@@ -3,7 +3,7 @@ AM_CXXFLAGS = $(WARNING_CXXFLAGS) $(PICKY_CXXFLAGS) $(HARDEN_CFLAGS) $(MISC_CXXF
 AM_LDFLAGS  = $(HARDEN_LDFLAGS)
 LDADD = ../crypto/libmoshcrypto.a ../network/libmoshnetwork.a ../statesync/libmoshstatesync.a ../terminal/libmoshterminal.a ../util/libmoshutil.a ../protobufs/libmoshprotos.a -lm $(TINFO_LIBS) $(protobuf_LIBS) $(CRYPTO_LIBS)
 
-mosh_server_LDADD = $(LDADD) $(LIBUTIL)
+mosh_server_LDADD = $(LDADD)
 
 bin_PROGRAMS =
 
diff --git a/src/tests/Makefile.am b/src/tests/Makefile.am
index 45446ac..a6f9fe6 100644
--- a/src/tests/Makefile.am
+++ b/src/tests/Makefile.am
@@ -63,7 +63,7 @@ nonce_incr_LDADD = ../network/libmoshnetwork.a ../crypto/libmoshcrypto.a ../util
 
 inpty_SOURCES = inpty.cc
 inpty_CPPFLAGS = -I$(srcdir)/../util
-inpty_LDADD = ../util/libmoshutil.a $(LIBUTIL)
+inpty_LDADD = ../util/libmoshutil.a
 
 clean-local: clean-local-check
 .PHONY: clean-local-check
-- 
2.12.2


From 036cbcfae56421c98dc6d32da38682eb86c2efe3 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sat, 8 Apr 2017 08:46:27 +0200
Subject: Fix tests on Haiku

The long options to locale are not standardized by POSIX, use the short
form, which is.

diff --git a/src/tests/e2e-test-subrs b/src/tests/e2e-test-subrs
index b85938d..8ec2fad 100644
--- a/src/tests/e2e-test-subrs
+++ b/src/tests/e2e-test-subrs
@@ -60,13 +60,13 @@ chr()
 set_locale()
 {
     # Test for a usable locale.
-    map=$(locale charmap 2>/dev/null)
+    map=$(locale -m 2>/dev/null)
     if [ "$map" = "utf-8" ] || [ "$map" = "UTF-8" ]; then
 	return 0
     fi
     # Attempt to find/set a usable locale.
     for i in en_US.UTF-8 en_US.utf8 C.UTF-8; do
-	map="$(env LANG=$i locale charmap 2>/dev/null)"
+	map="$(env LANG=$i locale -m 2>/dev/null)"
 	if [ "$map" = "utf-8" ] || [ "$map" = "UTF-8" ]; then
 	    export LANG=$i LC_ALL=''
 	    return 0
-- 
2.12.2

